search/lib/cmr.js

```diff
@@ -49,14 +50,28 @@ async function findCollections (params = {}) {
   return response.data.feed.entry;
 }
 
+async function stacIdToCmrCollectionId (providerId, stacId) {
```

Comment:
General comment: The code looks fine and is straightfoward, but none of the functions have any comments. We should add comments on functions and things like myCache to explain things like what the Stac cacheKey is and why we need to maintain this mapping between stack cacheKey and collection short_name and version. A sentence explaining what `convertParams` function does will make the code much more maintainable.

Reply:
Thanks for the comment @ygliuvt , good idea, I'll add in some comments.